doc/specs/stdlib_ascii.md

```diff
@@ -0,0 +1,149 @@
+---
+title: ASCII
+---
+
+# The `stdlib_ascii` module
+
+[TOC]
+
+## Introduction
+
+The `stdlib_ascii` module provides procedures for handling and manipulating
+intrinsic character variables and constants.
+
+
+## Constants provided by `stdlib_ascii`
+
+@note Specification of constants is currently incomplete.
+
+
+## Specification of the `stdlib_ascii` procedures
+
+@note Specification of procedures is currently incomplete.
+
+
+### `to_lower`
+
+#### Status
+
+Experimental
+
+#### Description
+
+Converts input character variable to all lowercase.
+
+#### Syntax
+
+```f90
+res = to_lower("HELLO!")
+! res == "hello!"
+```
+
+#### Class
+
+Pure function.
+
+#### Argument
+
+`string`: shall be an intrinsic character type. It is an `intent(in)` argument.
+
+#### Result value
+
+The result is an intrinsic character type of the same length as `string`.
+
+
+### `to_upper`
+
+#### Status
+
+Experimental
+
+#### Description
+
+Converts input character variable to all uppercase.
+
+#### Syntax
+
+```
+res = to_upper("hello!")
+! res == "HELLO!"
+```
+
+#### Class
+
+Pure function.
+
+#### Argument
+
+`string`: shall be an intrinsic character type. It is an `intent(in)` argument.
+
+#### Result value
+
+The result is an intrinsic character type of the same length as `string`.
+
+
+### `to_title`
+
+#### Status
+
+Experimental
+
+#### Description
+
+Returns capitalized version of input character variable.
+The first alphanumeric character is capitalized.
```

Comment:
Capitalization does not make sense for numeric characters. From the tests, alphabetic characters following numerals are not capitalized (e.g. `1st`).

On the other hand the first alphabetic character following a *non-alphanumeric* character will be capitalized.

I haven't really studied other languages to know, what are sensible title case rules for an input like `++c`. With the current implementation the result will be `++C`.

Reply:
The `capitalize` implementation I found so far in Fortran usually went for the first alphanumeric character. Without this experience I would have gone only for the first character of the string. It kind makes sense to check and skip leading delimiters or quotation marks, `++c` is borderline case, but returning `++C` feels consistent with my expectation on a `capitalize` function.

Reply:
I don't really have strong feelings about this. I think the most common use case is to capitalize individual words rather then sentences or even strings containing non-alpha numerical characters.

I still find the sentence `The first alphanumeric character is capitalized.` misleading. I understand the logic how it includes the ordinal numbers like `1st`, `2nd`, ... But dictionaries define capitalization only for letters, not for numerals.

Reply:
How about:

*The first alphanumerical character is transformed to uppercase if it is a letter, the rest of the character sequences is transformed to lowercase.*

Reply:
I would prefer using a negative form: *The first alphabetical character is transformed to uppercase unless it follows/succeeds a numeral. The rest of the character sequence is transformed to lowercase.*